hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastVersionLocator.java

```diff
@@ -132,26 +134,37 @@ private static File downloadFile(String url, File targetDirectory, String filena
             return targetFile;
         }
         HttpGet request = new HttpGet(url);
+
+        CloseableHttpResponse response = null;
+        FileOutputStream fos = null;
         try {
-            CloseableHttpResponse response = client.execute(request);
+            response = client.execute(request);
             if (response.getStatusLine().getStatusCode() != SC_OK) {
                 throw new GuardianException("Cannot download file from " + url
                         + ", http response code: " + response.getStatusLine().getStatusCode());
             }
             HttpEntity entity = response.getEntity();
-            FileOutputStream fos = new FileOutputStream(targetFile);
+            fos = new FileOutputStream(targetFile);
             entity.writeTo(fos);
-            fos.close();
             targetFile.deleteOnExit();
             return targetFile;
         } catch (IOException e) {
             throw rethrowGuardianException(e);
         } finally {
-            try {
-                client.close();
-            } catch (IOException e) {
-                // ignore
-            }
+            closeQuietly(fos);
+            closeQuietly(response);
+            closeQuietly(client);
+        }
+    }
+
+    private static void closeQuietly(Closeable closeable) {
```

Comment:
This method seems equivalent to `IOUtil.closeResource`

Reply:
fixed